actions/sync-calendar/gcal/gcal_test.go

```diff
@@ -0,0 +1,48 @@
+package gcal
+
+import (
+	"context"
+	"io/ioutil"
+	"testing"
+
+	"cloud.google.com/go/httpreplay"
+	"github.com/ghodss/yaml"
+	"github.com/spinnaker/rotation-scheduler/schedule"
+)
+
+const (
+	testCalendarID = "spinbot@spinnaker.io"
+)
+
+func TestSchedule(t *testing.T) {
```

Comment:
I'm not sure this test is really gaining us much here.  It looks like the replay library is only checking the URL of the requests, and not the content...so we're basically asserting that the following calls get made:
* 1 call to `POST /spinbot%40spinnaker.io/clear?alt=json\u0026prettyPrint=false`
* 4 calls  to `POST /spinbot%40spinnaker.io/events?alt=json\u0026prettyPrint=false\u0026sendUpdates=none`

So while I guess it is good to check that we are clearing the calendar and adding/updating 4 events, it feels like the most important parts of the logic (ie, that we're putting the right data in the events) are not being tested.  (I modified the logic locally to send the wrong dates/users for the events and the tests still passed.)

This might be good as an overall end-to-end or integration tests, but it feels like we're missing a more unit-test-like test on this module that tests that we're writing the correct data.

Reply:
Fair point - pulled out `calendar.Event` creation logic and added a test for that method.